### PR TITLE
Avalon SDK for direct mode Worker related API

### PR DESCRIPTION
**Pull request by karthikamurthy**
Signed-off-by: Karthika Murthy <karthika.murthy@intel.com>

### CODE REVIEW

**Comment by divyataori** [comment]
Please review this PR.

File: src/setup.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import os
+import sys
+import subprocess
+import re
+
+from setuptools import setup, find_packages
+
+setup(name='avalon_sdk_direct',
+      version='0.0.4',
```
Comment (rranjan3) [inline src/setup.py:25]: How are we deciding this version?

File: src/common/enums/worker.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    OFF_LINE = 2
+    DECOMMISSIONED = 3
+    COMPROMISED = 4
+    
+    @classmethod
+    def has_value(cls, value):
+        if value in cls._value2member_map_ :
+            return WorkerType(value)
+        if value in cls.__members__ :
+            return cls.value
```
Comment (rranjan3) [inline src/common/enums/worker.py]: Add a newline maybe. Will show up when part of some CI.

File: src/avalon_sdk_direct/unit_tests/test_worker_registry_jrpc_impl.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        self.__worker_id = res['result']['ids'][0]
+        self.__lookup_tag = res['result']['lookupTag']
+
+    def test_worker_lookup_next(self):
+        req_id = 17
+        logging.info(
+            'Calling worker_lookup_next with\n worker type %d\n' +
+            ' org_id %s\n app_ids %s\n lookUpTag %s\n',
+            self.__worker_type.value, self.__org_id, self.__app_ids, self.__lookup_tag)
+        res = self.__worker_registry_wrapper.worker_lookup_next(
```
Comment (rranjan3) [inline src/avalon_sdk_direct/unit_tests/test_worker_registry_jrpc_impl.py:124]: Can at least part of the response be verified if not throughout?

File: src/avalon_sdk_direct/unit_tests/test_worker_registry_jrpc_impl.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        logging.info(
+            'Calling test_worker_retrieve with\n worker_id %s\n',
+            self.__worker_id)
+        res = self.__worker_registry_wrapper.worker_retrieve(
+            self.__worker_id, req_id)
+        logging.info('Result: %s\n', res)
+        self.__org_id = res['result']['organizationId']
+        self.__app_ids =  res['result']['applicationTypeId']
+        self.__details = res['result']['details']
+        self.__status = res['result']['status']
```
Comment (rranjan3) [inline src/avalon_sdk_direct/unit_tests/test_worker_registry_jrpc_impl.py:97]: Assert or do not name the function as a test if that's not the intent.

File: src/avalon_sdk_direct/unit_tests/test_worker_registry_jrpc_impl.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+    def test_worker_set_status(self):
+        req_id = 14
+        self.__status = WorkerStatus.OFF_LINE
+        logging.info(
+            'Calling test_worker_set_status with\n worker_id %s\n status %d\n',
+            self.__worker_id, self.__status.value)
+        res = self.__worker_registry_wrapper.worker_set_status(
+            self.__worker_id, self.__status, req_id)
+        logging.info('Result: %s\n', res)
```
Comment (rranjan3) [inline src/avalon_sdk_direct/unit_tests/test_worker_registry_jrpc_impl.py:83]: Assertion needed.

File: src/Makefile [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+all :
+	python3 setup.py sdist bdist_wheel ;
+	#python3 setup_ethereum.py sdist bdist_wheel ;
```
Comment (rranjan3) [inline src/Makefile]: Do away with dead code.

**Review by rranjan3 — CHANGES_REQUESTED** [review]

File: src/setup.py (hunk 1 above)
Comment (karthikamurthy) [reply src/setup.py:25]: This is a dummy value for now. When need to automate the version before releasing the code as python sdk

File: src/Makefile (hunk 6 above)
Comment (karthikamurthy) [reply src/Makefile]: done

**Comment by karthikamurthy** [comment]
Code Merged in PR18

**Pull request closed without merging**